dsc/tests/dsc_assertion.tests.ps1

```diff
@@ -0,0 +1,20 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+Describe 'Assertion resource tests' -Skip:(!$IsWindows) {
+    It 'Example works for <operation>' -TestCases @(
+        @{ operation = 'get' }
+        @{ operation = 'set' }
+        @{ operation = 'test' }
+# TODO: Add export to test when https://github.com/PowerShell/DSC/issues/428 is fixed
+#        @{ operation = 'export' }
+    ) {
+        param($operation)
+        $jsonPath = Join-Path $PSScriptRoot '../examples/assertion.dsc.yaml'
+        $out = dsc config $operation -f $jsonPath 2> "$TestDrive/trace.log"
+        $LASTEXITCODE | Should -Be 2
```

Comment:
Why is it expected that the exit code will be 2 i.e. `EXIT_DSC_ERROR` instead of 8 i.e. `EXIT_DSC_ASSERTION_FAILED` here?

Reply:
You're right, this should be 8

Reply:
Actually, the 8 is coming from the child `dsc` process, but the parent one is returning `2` is which the DSC Error code.  So this is correct as we don't propagate the child exit code to the parent.

Reply:
Created https://github.com/PowerShell/DSC/issues/737 as it's not a simply change, but something we should eventually support